### PR TITLE
Refactor to use discovery-types package

### PR DESCRIPTION
**Pull request by michalsidzej**
Resolves L2B-2174

Context: we want to move `@l2beat/discoverry` package to https://github.com/l2beat/tools repo. This requires extracting discovery types so they can be used in all the relevant packages. This PR is the last step before moving discovery package.

### CODE REVIEW

**Comment by lucadonnoh** [comment]
@michalsidzej could you give a small description for the reason behind this change? just curious :p

**Comment by michalsidzej** [comment]
> @michalsidzej could you give a small description for the reason behind this change? just curious :p

@lucadonnoh provided more context in PR description

File: packages/discovery/src/discovery/config/RawDiscoveryConfig.ts [hunk 1]
```diff
@@ -12,7 +8,7 @@ export type DiscoveryContract = z.infer<typeof DiscoveryContract>
 export const DiscoveryContract = z.object({
   ignoreDiscovery: z.optional(z.boolean()),
   proxyType: z.optional(ManualProxyType),
-  ignoreInWatchMode: z.optional(z.array(z.string())),
+  ignoreInWatchMode: z.z.optional(z.array(z.string())),
```
Comment (sdlyy) [inline packages/discovery/src/discovery/config/RawDiscoveryConfig.ts:11]: Does it provide some form of stronger validation - the double z? 😄

**Review by sdlyy — APPROVED** [review]
Looks fine, I'll do the second run later on

File: packages/discovery/src/discovery/config/RawDiscoveryConfig.ts (hunk 1 above)
Comment (michalsidzej) [reply packages/discovery/src/discovery/config/RawDiscoveryConfig.ts:11]: I wonder why does TS not shout about it

Comment (michalsidzej) [reply packages/discovery/src/discovery/config/RawDiscoveryConfig.ts]: 7948981e

**Pull request merged**